include/ck_tile/ops/gemm/kernel/gemm_kernel.hpp

```diff
@@ -236,56 +296,119 @@ struct GemmKernel
-            {
-                return make_naive_tensor_view<address_space_enum::global>(
-                    c_start,
-                    make_tuple(kargs.M, kargs.N),
-                    make_tuple(1, kargs.stride_C),
-                    number<1>{},
-                    number<1>{});
-            }
-        }();
+        if(kargs.KBatch == 1)
+        {
+            auto c_tensor_view = [&]() {
+                if constexpr(std::is_same_v<CLayout, tensor_layout::gemm::RowMajor>)
+                {
+                    return make_naive_tensor_view<address_space_enum::global>(
+                        c_start,
+                        make_tuple(kargs.M, kargs.N),
+                        make_tuple(kargs.stride_C, 1),
+                        number<GemmPipeline::VectorSizeC>{},
+                        number<1>{});
+                }
+                else
+                {
+                    return make_naive_tensor_view<address_space_enum::global>(
+                        c_start,
+                        make_tuple(kargs.M, kargs.N),
+                        make_tuple(1, kargs.stride_C),
+                        number<1>{},
+                        number<1>{});
+                }
+            }();
 
-        auto c_pad_view = [&]() {
-            if constexpr(std::is_same_v<CLayout, tensor_layout::gemm::RowMajor>)
-            {
-                return pad_tensor_view(
-                    c_tensor_view,
-                    make_tuple(number<TilePartitioner::kM>{}, number<TilePartitioner::kN>{}),
-                    sequence<false, GemmPipeline::kPadN>{});
-            }
-            else
+            auto c_pad_view = [&]() {
+                if constexpr(std::is_same_v<CLayout, tensor_layout::gemm::RowMajor>)
+                {
+                    return pad_tensor_view(
+                        c_tensor_view,
+                        make_tuple(number<TilePartitioner::kM>{}, number<TilePartitioner::kN>{}),
+                        sequence<false, GemmPipeline::kPadN>{});
+                }
+                else
+                {
+                    return pad_tensor_view(
+                        c_tensor_view,
+                        make_tuple(number<TilePartitioner::kM>{}, number<TilePartitioner::kN>{}),
+                        sequence<GemmPipeline::kPadM, false>{});
+                }
+            }();
+            auto CBlockWindow_pad = make_tile_window(
+                c_pad_view,
+                make_tuple(number<TilePartitioner::kM>{}, number<TilePartitioner::kN>{}),
+                {i_m, i_n});
+            EpiloguePipeline{}(CBlockWindow_pad, c_block_tile);
+        }
+        else
+        {
+            auto c_tensor_view = [&]() {
+                if constexpr(std::is_same_v<CLayout, tensor_layout::gemm::RowMajor>)
+                {
+                    return make_naive_tensor_view<address_space_enum::global,
+                                                  memory_operation_enum::atomic_add>(
```

Comment:
What about some refactoring this part? You just change memory operation, the rest is left unchanged. You may create some even lambda function which takes enum op and returns appropriate tensor view.

Reply:
Yes, we should to do this but maybe it is better to left it for @aledudek since he works on refactor of this part?